### PR TITLE
Cancel message edit if content remains unchanged

### LINKED ISSUE #3344: Check edit content before sending update
As reported on the public room

`When editing a message, if no change is made and the send button is tapped, it still sends it as an edit rather than canceling the edit.`

### PR DESCRIPTION
**Pull request by thepearl**
### Pull Request Checklist

When a user attempts to send an edit without modifying the message content, the edit operation will now be cancelled. 
This prevents sending an identical message, avoids unnecessary "edited" indicators, and ensures that any volatile drafts are restored correctly as if the edit was explicitly cancelled

closes https://github.com/element-hq/element-x-ios/issues/3344

| BEFORE | AFTER |
| --- | --- |
| before-unchanged-edit-marked-edited | image |

PS for maintainers: if you have any edge cases please let me know i can test it using the newly added test case (or checkout to my branch and test it directly) 

- [x] I read the [contributing guide](https://github.com/element-hq/element-x-ios/blob/develop/CONTRIBUTING.md).
- [x] I am aware of the [etiquette](https://github.com/element-hq/element-x-ios/blob/develop/CONTRIBUTING.md#etiquette).
- [ ] Pull request contains a [changelog label](https://github.com/element-hq/element-x-ios/blob/develop/CONTRIBUTING.md#changelog).
- [x] This PR has been made with the help of an LLM.

**UI changes have been tested with:**
- [x] iPhone and iPad simulators in portrait and landscape orientations.
- [ ] Dark mode enabled and disabled.
- [ ] Various sizes of dynamic type.
- [ ] Voiceover enabled.

### CODE REVIEW

File: ElementX/Sources/Screens/RoomScreen/ComposerToolbar/ComposerToolbarViewModel.swift [hunk 1]
```diff
@@ -265,6 +280,10 @@ final class ComposerToolbarViewModel: ComposerToolbarViewModelType, ComposerTool
             } else {
                 set(text: plainText)
             }
+            
+            if state.composerMode.isEdit {
+                originalEditContent = currentContent
+            }
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/RoomScreen/ComposerToolbar/ComposerToolbarViewModel.swift:286]: ~~Why do this here instead of in `set(mode: ComposerMode)`  instead of just doing `originalEditContent = nil` there?~~

Nevermind, it's a stupid timing issue.

File: ElementX/Sources/Screens/RoomScreen/ComposerToolbar/ComposerToolbarViewModel.swift [hunk 2]
```diff
@@ -453,6 +472,16 @@ final class ComposerToolbarViewModel: ComposerToolbarViewModelType, ComposerTool
                                          intentionalMentions: .init(userIDs: plainComposerContent.mentionedUserIDs, atRoom: plainComposerContent.containsAtRoomMention)))
     }
     
+    private func cancelEdit() {
```
Comment (stefanceriu) [inline ElementX/Sources/Screens/RoomScreen/ComposerToolbar/ComposerToolbarViewModel.swift:475]: 👌

**Review by stefanceriu — APPROVED** [review]
Looks good to me, thanks for taking the time to fix it!

**Comment by stefanceriu** [comment]
> thepearl force-pushed the fix/edit-unchanged-message-cancels branch from c159b1a to 54ff95f 2 minutes ago

It was good to go and you just ruined it, what gives?

**Comment by thepearl** [comment]
@stefanceriu i rebased the branch onto develop

**Comment by stefanceriu** [comment]
> @stefanceriu i rebased the branch onto develop

Ah sorry, the stupid comparison screen led me astray.
